clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderProducerCallback.java

```diff
@@ -165,7 +165,26 @@ public void onCompletion(RecordMetadata recordMetadata, Exception e) {
             producedRecordNum++;
             producedRecordSize += chunkKey.remaining() + chunkValue.remaining();
           }
-
+          if (chunkedRmdManifest != null) {
+            for (int i = 0; i < chunkedRmdManifest.keysWithChunkIdSuffix.size(); i++) {
+              ByteBuffer chunkKey = chunkedRmdManifest.keysWithChunkIdSuffix.get(i);
+              ByteBuffer chunkValue = rmdChunks[i];
+              Put chunkPut = instantiateRmdChunkPut();
+              chunkPut.schemaId = schemaId;
+              chunkPut.replicationMetadataPayload = chunkValue;
+              LeaderProducedRecordContext producedRecordForChunk =
+                  LeaderProducedRecordContext.newChunkPutRecord(ByteUtils.extractByteArray(chunkKey), chunkPut);
+              producedRecordForChunk.setProducedOffset(-1);
+              ingestionTask.produceToStoreBufferService(
+                  sourceConsumerRecord,
+                  producedRecordForChunk,
+                  subPartition,
+                  kafkaUrl,
+                  beforeProcessingRecordTimestamp);
+              producedRecordNum++;
+              producedRecordSize += chunkKey.remaining() + chunkValue.remaining();
```

Comment:
Hmm the 2 primitive long variable is a bit tricky to handle. I can think of creating a local nested class to wrap it but it might not be necessary.. Could you please take a look at what I do in my latest iteration and see if it makes sense.